MdeModulePkg/Library/UefiBootManagerLib/BmBoot.c

```diff
@@ -2436,6 +2436,241 @@ BdsLibGetBootableHandle (
   return ReturnHandle;
 }
 
+typedef struct {
+  CHAR16* FileName;
+  CHAR16* BootOptionFmt;
+} PRE_INSTALLED_BOOT_OPT;
+
+PRE_INSTALLED_BOOT_OPT PreInstalledBootOpts[] = {
+  { L"\\EFI\\Microsoft\\Boot\\bootmgfw.efi", L"Windows Boot Manager (on %s)" },
+  { L"\\EFI\\Suse\\elilo.efi",               L"Suse Boot Manager (on %s)" },
+  { L"\\EFI\\Redhat\\elilo.efi",             L"RedHat Boot Manager (on %s)" },
+};
+
+PRE_INSTALLED_BOOT_OPT PreInstalledBootOptsLnx[] = {
+  { L"\\EFI\\ubuntu\\shimx64.efi",           L"Ubuntu (on %s)" },
+  { L"\\EFI\\ubuntu\\grubx64.efi",           L"Ubuntu (on %s)" },
+  { L"\\EFI\\redhat\\shimx64.efi",           L"RedHat (on %s)" },
+  { L"\\EFI\\redhat\\grubx64.efi",           L"RedHat (on %s)" },
+  { L"\\EFI\\fedora\\shimx64.efi",           L"Fedora (on %s)" },
+  { L"\\EFI\\fedora\\grubx64.efi",           L"Fedora (on %s)" },
+  { L"\\EFI\\centos\\shimx64.efi",           L"CentOS (on %s)" },
+  { L"\\EFI\\centos\\grubx64.efi",           L"CentOS (on %s)" },
+  { L"\\EFI\\opensuse\\shimx64.efi",         L"OpenSuse (on %s)" },
+  { L"\\EFI\\opensuse\\grubx64.efi",         L"OpenSuse (on %s)" },
+  { L"\\EFI\\debian\\shimx64.efi",           L"Debian (on %s)" },
+  { L"\\EFI\\debian\\grubx64.efi",           L"Debian (on %s)" },
+};
+
+PRE_INSTALLED_BOOT_OPT DtsBootOpt = {
+  L"\\EFI\\DTS\\grubx64.efi",           L"Ubuntu (on %s)" 
+};
+
+EFI_BOOT_MANAGER_LOAD_OPTION *
+CreatePreInstalledBootOption (
+  IN OUT EFI_BOOT_MANAGER_LOAD_OPTION   *BootOptions,
+  IN OUT UINTN                          *BootOptionCount,
+  IN EFI_HANDLE                         Handle,
+  IN PRE_INSTALLED_BOOT_OPT             *BootOpt

```

Comment:
nit:
```suggestion
  IN CONST PRE_INSTALLED_BOOT_OPT       *BootOpt
```
Otherwise it looks like it can be in/out.

Reply:
Done: https://github.com/Dasharo/edk2/pull/70/commits/d9d411152e4b6244f44d4f5af663b63145b6e4f6#diff-d2820fe49b39a17c1aa46361e3d22ee285be3bd600dae98c0dc36d3bf030d04fR2534